manual_tag_info_gather/manual_tag_info_collector.py

```diff
@@ -0,0 +1,75 @@
+"""
+For all users assigned tagging work, check how many tags are left, and how many require pruning.
+Also, generate an aggregate ecosystem_package_topic_map for PGM training.
+"""
+"""
+To run the script:
+Add the config parameters to env
+cp config.py.template config.py
+All the user_name as user_name = ["name1", "name2",...]
+python manual_tag_info_collector.py
+"""
+
+import config
+import requests
+import json
+import sys
+
+# Update variables from config.
+user_names = config.user_names
+prefix_url = config.prefix_url
+ecosystem = config.ecosystem
+Authorization = config.Authorization
+
+header = {'Authorization': Authorization}
+user_data_not_found = []
+final_data_list = []
+for_each_ecosystem = {'ecosystem': ecosystem}
+final_package_dict = {}
+
+# For each user, collect the package_topic_map list
+try:
+    for name in user_names:
+        url = prefix_url + name + "/" + ecosystem
+        response = requests.get(url, headers=header)
+        if response.status_code != 200:
+            user_data_not_found.append(name)
+            continue
+        response_data = json.loads(response.content)
+        package_topic_map = response_data.get(
+            'data', {}).get('package_topic_map', {})
+        for package_name, tag_list in package_topic_map.items():
+            if package_name in final_package_dict.keys():
+                tag_list = tag_list.extend(final_package_dict[package_name])
+            final_package_dict[package_name] = list(set(tag_list))
+except Exception as e:
+    raise e
+
+if len(user_data_not_found) == len(user_names):
+    print("Request unsuccessfull for all users")
+    sys.exit(1)
+
+if len(user_data_not_found) > 0:
+    print("Request unsuccessfull for following users %s" % user_data_not_found)
+
+
+for_each_ecosystem['package_topic_map'] = final_package_dict
+final_list.append(for_each_ecosystem)
```

Comment:
`final_list` is unresolved reference (not sure how you were able to run the script with this). Do you mean `final_data_list` ?